core/clustersmngr/factory.go

```diff
@@ -589,13 +592,17 @@ func (cf *clustersManager) userNsList(ctx context.Context, user *auth.UserPrinci
 		if err != nil {
 			// This may not completely fail the request, e.g. if some of the clusters
 			// are able to respond with their namespaces. So log the error and continue.
-			cf.log.Error(err, "failed updating namespaces from user client")
+			cf.log.Error(err, "Error updating namespaces from user client", "user", user)
 		}
 	}
 
 	cf.UpdateUserNamespaces(ctx, user)
 
-	return cf.GetUserNamespaces(user)
+	userNamespaces = cf.GetUserNamespaces(user)
+
+	cf.log.V(logger.LogLevelDebug).Info("Updated namespace access cache for user", "userNamespaces", clusterNamespacesLogValue(userNamespaces), "user", user, "ttl", userNamespaceTTL.String())
```

Comment:
This reveals sensitive information (user names) in the logs. I don't know if this is even a problem but wanted to make sure this is a deliberate decision. 🙂

Reply:
Interesting points!
Likely the group is the more relevant information for debugging. We should start with the group instead of the user. Besides that, its probably a great idea to have somewhere in the docs a note about what information related to the user we log.

Reply:
https://github.com/weaveworks/weave-gitops/issues/3318